spec.html

```diff
@@ -24176,16 +24314,265 @@ <h1>Runtime Semantics: Evaluation</h1>
+        <dd>Creates a function for a decorator's context object which can be used to add an extra initializer to the class.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _addInitializerClosure_ be a new Abstract Closure with parameters (_initializer_) that captures _initializers_ and _decorationState_ and performs the following steps when called:
+          1. If _decorationState_.[[Finished]] is *true*, throw a *TypeError* exception.
+          1. Append _initializer_ to _initializers_.
+          1. Return *undefined*.
+        1. Return CreateBuiltinFunction(_addInitializerClosure_, 1, *""*, &laquo; &raquo;).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createdecoratorcontextobject" type="abstract operation">
+      <h1>
+        CreateDecoratorContextObject (
+          _kind_: ~class~, ~field~, ~method~, ~accessor~, ~getter~, or ~setter~,
+          _name_: a String, a Symbol, or Private Name,
+          _initializers_: a List of function objects,
+          _decorationState_: a Record with a field [[Finished]] (a Boolean),
+          optional _isStatic_: a Boolean,
+        ): an Object
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It returns a context object which is passed as the second argument to decorators of the given class element.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _contextObj_ be OrdinaryObjectCreate(%Object.prototype%).
+        1. If _kind_ is ~method~, let _kindStr_ be *"method"*.
+        1. Else if _kind_ is ~getter~, let _kindStr_ be *"getter"*.
+        1. Else if _kind_ is ~setter~, let _kindStr_ be *"setter"*.
+        1. Else if _kind_ is ~accessor~, let _kindStr_ be *"accessor"*.
+        1. Else if _kind_ is ~field~, let _kindStr_ be *"field"*.
+        1. Else,
+          1. Assert: _kind_ is ~class~.
+          1. Let _kindStr_ be *"class"*.
+        1. Perform ! CreateDataPropertyOrThrow(_contextObj_, *"kind"*, _kindStr_).
+        1. If _kind_ is not ~class~, then
+          1. Perform ! CreateDataPropertyOrThrow(_contextObj_, *"access"*, CreateDecoratorAccessObject(_kind_, _name_)).
+          1. If _isStatic_ is present, perform ! CreateDataPropertyOrThrow(_contextObj_, *"isStatic"*, _isStatic_).
+          1. If _name_ is a Private Name, then
+            1. Perform ! CreateDataPropertyOrThrow(_contextObj_, *"isPrivate"*, *true*).
+            1. Perform ! CreateDataPropertyOrThrow(_contextObj_, *"name"*, _name_.[[Description]]).
+          1. Else,
+            1. Perform ! CreateDataPropertyOrThrow(_contextObj_, *"isPrivate"*, *false*).
+            1. Perform ! CreateDataPropertyOrThrow(_contextObj_, *"name"*, _name_).
+        1. Else,
+          1. Perform ! CreateDataPropertyOrThrow(_contextObj_, *"name"*, _name_).
+        1. Let _addInitializer_ be CreateAddInitializerFunction(_initializers_, _decorationState_).
+        1. Perform ! CreateDataPropertyOrThrow(_contextObj_, *"addInitializer"*, _addInitializer_).
```

Comment:
Yes this was a mistake, there is an open issue about it here: https://github.com/tc39/proposal-decorators/issues/464

So far the consensus seems to be that the spec does not need to change and the explainer can be updated. I think this may make the most sense.